### DISCUSSION TITLE
File has different text color

### DISCUSSION THREAD

**Discussion by timii** [nvim-tree/nvim-tree.lua#1680] [Q&A] [answered]
Hi,

I currently have this weird issue, that specifically my .js files have a different text color than all the other file types:
![image](https://user-images.githubusercontent.com/56680452/197417376-e768f4af-7a64-4c54-b234-d4094240ee91.png)

I don't think that I have changed any config option that would modify the text color of just .js files. My nvim-tree config for reference:
```lua
require("nvim-tree").setup({
    -- Base
    auto_reload_on_write = true,
    update_focused_file = {
		  enable = true,
	},

    -- remove default mappings
    remove_keymaps = {
        'f' -- disable live filtering and be able to use telescope keymap
    },

    -- always open tree automatically
    open_on_setup = true,
    open_on_setup_file = true,
    --ignore_buffer_on_setup = true,
    create_in_closed_folder = true,
    open_on_tab = true,

    -- keep cursor on first letter of the filename 
    hijack_cursor = true,
    
    -- show diagnostics
    diagnostics = {
	  	enable = true,
	  	show_on_dirs = true,
        icons = {
			  hint = "",
			  info = "",
			  warning = "",
			  error = "",
		},
	},

    -- Git integration
    git = {
      ignore = false
    },

    -- window setup
    view = {
      -- custom key mappings
      mappings = {
        list = {
          -- toggle focus back to current file
          { key = 'tb', action = function() vim.cmd.wincmd 'p' end }
        }
      }
    },

    -- UI renderer setup
    renderer = {
      highlight_git = true,
      root_folder_modifier = ":t",
      icons = {
			  glyphs = {
				  git = {
					  unstaged = "M",
					  staged = "S",
					  unmerged = "",
					  renamed = "U",
					  untracked = "U",
					  deleted = "D",
					  ignored = "◌",
				  },
			  },
		  },
    },
})
```

If any other information is needed, I will gladly add it to my question or reply with it in the comments. 

Thanks.

**Comment by alex-courtis** [comment]
There are some highlight groups that change the colours such as `NvimTreeExecFile`, which may be the case here.

See `:help nvim-tree-highlight`.

You can "diagnose" highlighting issues by looking at actual highlight groups: `:so $VIMRUNTIME/syntax/hitest.vim`

**Reply by timii** [reply]
After running `:so $VIMRUNTIME/syntax/hitest.vim` it looks like `NvimTreeExecFile` has the same color as the .js files in the tree. So I tried changing the color of the `NvimTreeExecFile` highlight group with this line: 

`vim.api.nvim_set_hl(0, 'NvimTreeExecFile', { fg = "#ffffff" })`

But it didn't change the text color. I even tried the same, but changing the highlight group colors of `NvimTreeBookmark`, `NvimTreeGitStaged` and `NvimTreeOpenedFiled`, but none of them changed the text color of the .js files.

I also used `:highlight NvimTreeExecFile` to get the options that were changed for the highlight group. Does it maybe have something to do with the highlight group using `guifg` instead of `fg` and that's why my changes didn't do anything? I tried setting `guifg=#ffffff`, but it just threw an error saying that guifg is an invalid key.

Another question is, why the .js files were even seen as executables? I checked how the text color looked for other file types and .exe files do have the same color, which makes sense, but other file types I tested didn't. What decides what coloring different file types have, and can I maybe just exclude .js files from being seen as executables?

**Reply by alex-courtis** [reply]
> So I tried changing the color of the `NvimTreeExecFile` highlight group with this line:
> 
> `vim.api.nvim_set_hl(0, 'NvimTreeExecFile', { fg = "#ffffff" })`

I tried the above (with #ff0000) with a clean setup and the colours of the executable files in the tree were changed to red.

`:so $VIMRUNTIME/syntax/hitest.vim` confirmed the change for that group. Did you see that confirmation?

Please also test with clean nvt-min.lua as per [bug report](https://github.com/nvim-tree/nvim-tree.lua/issues/new?assignees=&labels=bug&template=bug_report.yml). There may be other plugins or configuration that are interfering.

> I also used :highlight NvimTreeExecFile to get the options that were changed for the highlight group. Does it maybe have something to do with the highlight group using guifg instead of fg and that's why my changes didn't do anything? I tried setting guifg=#ffffff, but it just threw an error saying that guifg is an invalid key.

Please confirm that you have `termguicolors` set as per `:help nvim-tree-highlight`.

> Another question is, why the .js files were even seen as executables?

Are the files actually executable? `ls -l`

**Reply by timii** [reply]
> I tried the above (with #ff0000) with a clean setup and the colours of the executable files in the tree were changed to red.

I tried the same with #ff0000 as well. I just saved it and ran `luafile %` and I saw that, using `:so $VIMRUNTIME/syntax/hitest.vim`, that NvimTreeExecFile actually changed color. But after restarting and checking the highlight groups again, NvimTreeExecFile was again back to the default color.
I also tested it using nvt-min.lua and successfully changed the text color and it didn't reset to the default colors after restarting.

> Please confirm that you have termguicolors set as per :help nvim-tree-highlight.

I checked, and I have set `vim.opt.termguicolors = true` in my init.lua

> Are the files actually executable? ls -l

I checked and the .js files looked exactly the same as the other filetypes:
Using ubuntu `ls -l`:
![image](https://user-images.githubusercontent.com/56680452/198840844-3a915f7a-7f5a-433b-96ed-1ae5f8683a64.png)

Using powershell `ls`:
![image](https://user-images.githubusercontent.com/56680452/198840882-6c1424b3-8d89-43ea-b842-290a34750e2b.png)

**Reply by alex-courtis** [reply]
It looks like the files are actually executable, so nvim-tree is simply reporting what it sees.

Remove the executable attribute:
`find . -type f -exec chmod 644 {} \;`

You'll need to consult your windows documentation on filesystems and permissions if that doesn't work.

**Reply by timii** [reply]
I tried changing it using `chmod` but the color is still the same. 

> Please also test with clean nvt-min.lua as per [bug report](https://github.com/nvim-tree/nvim-tree.lua/issues/new?assignees=&labels=bug&template=bug_report.yml). There may be other plugins or configuration that are interfering.

In this comment from before, you mentioned that it could maybe be a different plugin interfering, but what other plugin would set or reset the value of the NvimTreeExecFile highlight group, when no other plugin changes or configures nvim-tree? Do you maybe a guess what plugin or what kind of plugin would modify or overwrite NvimTreeExecFile?

**Reply by timii** [reply]
Seems like I found a solution. After testing out a few other plugin files where the highlight group could be overwritten, I found out that my theme (`gruvbox`) was overriding the NvimTreeExecFile value for whatever reason. After adding the following line right after setting my theme, the new text color value was not overwritten after restarting:
```lua
vim.api.nvim_set_hl(0, 'NvimTreeExecFile', { fg = "#ebdbb2" }
```

@alex-courtis thanks for all your help and the idea that it could have something to do with a different plugin. I will also add my solution as a separate answer, so I can mark it as the answer.

**Comment by timii** [comment, marked as answer]
Seems like I found a solution. After testing out a few other plugin files where the highlight group could be overwritten, I found out that my theme (`gruvbox`) was overriding the NvimTreeExecFile value for whatever reason. After adding the following line right after setting my theme, the new text color value was not overwritten after restarting:
```lua
vim.api.nvim_set_hl(0, 'NvimTreeExecFile', { fg = "#ebdbb2" }
```

**Reply by alex-courtis** [reply]
> overriding the NvimTreeExecFile value for whatever reason

That is very strange! `ExecFile` is not linked to any other groups so would require something to explicily set it.

**Reply by timii** [reply]
That's what I thought and why I was wondering how any other plugin could interfere, but maybe it's just `gruvbox` and every other theme is fine.

**Reply by pleshevskiy** [reply]
https://github.com/ellisonleao/gruvbox.nvim/blob/main/lua/gruvbox/groups.lua#L331
Many themes are overridden for other plugins, but this one is over the top)

**Reply by timii** [reply]
Looks like we found the problem, and it really was just the theme. Thanks for the look into the gruvbox repo and confirming the assumption.

**Reply by alex-courtis** [reply]
Thanks all for digging into this one.
